src/subscription_manager/utils.py

```diff
@@ -62,6 +62,77 @@ def __repr__(self):
         return pprint.pformat(self.as_dict())
 
 
+def three_way_merge(local=None, base=None, remote=None, on_conflict="remote"):
+    """
+    Performs a three-way merge on the local and remote dictionaries with a given base.
+    :param local: The dictionary of the current local values
+    :param base: The dictionary with the values we've last seen
+    :param remote: The dictionary with "their" values
+    :param on_conflict: Either "remote" or "local" or None. If "remote", the remote changes
+                               will win any conflict. If "local", the local changes will win any
+                               conflict. If anything else, an error will be thrown.
+
+    :return: The dictionary of values as merged between the three provided dictionaries.
+    """
+    # Ensure that we at least have a value for each keyword arg. Left each input param as a keyword
+    # arg as it improves readability
+
+    result = {}
+
+    local = local or {}
+    base = base or {}
+    remote = remote or {}
+
+    if on_conflict == "remote":
+        winner = remote
+    elif on_conflict == "local":
+        winner = local
+    else:
+        raise ValueError('on_conflict can be remote/local')
+
+    all_keys = sorted(list(set(local.keys()) | set(base.keys()) | set(remote.keys())))
+
+    for key in all_keys:
+
+        local_changed = detect_changed(base=base, other=local, key=key)
+        remote_changed = detect_changed(base=base, other=remote, key=key)
+
+        if local_changed == remote_changed:
+            if key in winner:
+                result[key] = winner[key]
+        elif remote_changed:
+            if key in remote:
+                result[key] = remote[key]
+        elif local_changed:
+            if key in local:
+                result[key] = local[key]
+
+    return result
+
+
+def detect_changed(base=None, other=None, key=None):
```

Comment:
Same argument here: you don't need to provide defaults, but can still call with kwargs for readability.